rust/src/lib/nm/profile_search.rs

```diff
@@ -0,0 +1,367 @@
+            .get(&(
+                base_iface.name.to_string(),
+                NmIfaceType::from(&base_iface.iface_type),
+            ))
+            .map(Rc::as_ref)
+    }
+
+    /// Prefer activated in the order of:
+    /// * Currently activated
+    /// * Biggest `connection.autoconnect-priority`
+    /// * Biggest `connection.timestamp`
+    /// * Biggest `connection.uuid`
+    pub(crate) fn get_prefered_saved<'s>(
+        &'s self,
+        name: &str,
+        nm_iface_type: &NmIfaceType,
+    ) -> Option<&'s NmConnection> {
+        let nm_conns = self
+            .nm_saved_conns_index
+            .get(&(name.to_string(), nm_iface_type.clone()))?;
+
+        for nm_conn in nm_conns {
+            if let Some(uuid) = nm_conn.uuid() {
+                if self.nm_acs.contains_key(uuid) {
+                    return Some(Rc::as_ref(nm_conn));
+                }
+            }
+        }
+
+        let mut nm_conns: Vec<&NmConnection> =
+            nm_conns.iter().map(|n| n.as_ref()).collect();
+        nm_conns.sort_unstable_by(|a, b| {
+            nm_conn_activation_sort_keys(a)
+                .cmp(&nm_conn_activation_sort_keys(b))
+        });
+
+        nm_conns.pop()
+    }
+
+    /// Get all connections can be used to activate specified interface
+    pub(crate) fn get_saved(
+        &self,
+        base_iface: &BaseInterface,
+    ) -> Vec<&NmConnection> {
+        let mut ret: Vec<&NmConnection> = if let Some(nm_conns) =
+            self.nm_saved_conns_index.get(&(
+                base_iface.name.to_string(),
+                NmIfaceType::from(&base_iface.iface_type),
+            )) {
+            nm_conns.iter().map(Rc::as_ref).collect()
+        } else {
+            Vec::new()
+        };
+        if base_iface.identifier == Some(InterfaceIdentifier::MacAddress) {
+            if let Some(mac) = base_iface.mac_address.as_deref() {
+                if let Some(nm_conns) = self.nm_saved_conns_mac.get(&(
+                    mac.to_uppercase(),
+                    NmIfaceType::from(&base_iface.iface_type),
+                )) {
+                    for nm_conn in nm_conns {
+                        ret.push(nm_conn);
+                    }
+                }
+            }
+        }
+
+        ret.sort_unstable_by(|a, b| a.uuid().cmp(&b.uuid()));
+        ret.dedup();
+        ret
+    }
+
+    pub(crate) fn get_nm_ac(
+        &self,
+        iface: &BaseInterface,
+    ) -> Option<&NmActiveConnection> {
+        self.nm_acs_index
+            .get(&(
+                iface.name.to_string(),
+                NmIfaceType::from(&iface.iface_type),
+            ))
+            .map(Rc::as_ref)
+    }
+
+    pub(crate) fn saved_iter(&self) -> impl Iterator<Item = &NmConnection> {
+        self.nm_saved_cons.values().map(Rc::as_ref)
+    }
+
+    pub(crate) fn is_uuid_activated(&self, uuid: &str) -> bool {
+        self.nm_acs.contains_key(uuid)
+    }
+
+    pub(crate) fn get_applied_by_name_type(
```

Comment:
Suggestion: get_applied_by_iface

Reply:
We already have `get_applied()` which takes `BaseInterface`, so this `by_iface` might be not fit here.